From a6a8c7f04e9ec2c422880df5190456e27249d3df Mon Sep 17 00:00:00 2001
From: Zhu Honglei <zhuhonglei1714@phytium.com.cn>
Date: Mon, 17 Jun 2024 19:33:10 +0800
Subject: [PATCH 118/150] dt-bindings: dma: phytium,ddma: Add bindings for
 Phytium DDMA

This patch document the DT bindings for the Phytium DDMA controller.

Signed-off-by: Zhu Honglei <zhuhonglei1714@phytium.com.cn>
Signed-off-by: Li Mingzhe <limingzhe1839@phytium.com.cn>
Signed-off-by: Wang Yinfeng <wangyinfeng@phytium.com.cn>
Change-Id: I717de8cba5a1a689a264664aade4762dc34f4781
Signed-off-by: Andrew Powers-Holmes <aholmes@omnom.net>
---
 .../devicetree/bindings/dma/phytium,ddma.yaml | 58 +++++++++++++++++++
 MAINTAINERS                                   |  1 +
 2 files changed, 59 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/dma/phytium,ddma.yaml

diff --git a/Documentation/devicetree/bindings/dma/phytium,ddma.yaml b/Documentation/devicetree/bindings/dma/phytium,ddma.yaml
new file mode 100644
index 000000000000..b7e6b7738a30
--- /dev/null
+++ b/Documentation/devicetree/bindings/dma/phytium,ddma.yaml
@@ -0,0 +1,58 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+title: Phytium DDMA Controller bindings
+
+description:
+  The Phytium DDMA is a general-purpose direct memory access
+  controller capable of supporting 8 independent DMA channels.
+  Each channel can have up to 32 requests. DMA clients connected
+  to the Phytium DDMA controller must use the format described
+  in the dma.txt file, using a two-cell specifier for each.
+
+channel:
+  a phandle to the DMA controller plus the following two integer cells:
+    1. The channel id
+    2. The request line number
+
+maintainers:
+  - Huang Jie <huangjie1663@phytium.com.cn>
+
+allOf:
+  - $ref: "dma-controller.yaml#"
+
+properties:
+  "#dma-cells":
+    const: 2
+
+  compatible:
+    const: phytium,ddma
+
+  reg:
+    maxItems: 1
+
+  interrupts:
+    maxItems: 1
+
+  dma-channels:
+    minItems: 1
+    maxItems: 8
+    description: it indicates that the number of channels are used
+
+required:
+  - compatible
+  - reg
+  - interrupts
+  - dma-channels
+
+unevaluatedProperties: false
+
+examples:
+    ddma0: ddma@28003000 {
+        compatible = "phytium,ddma";
+        reg = <0x0 0x28003000 0x0 0x1000>;
+        interrupts = <GIC_SPI 75 IRQ_TYPE_LEVEL_HIGH>;
+        #dma-cells = <2>;
+        dma-channels = <8>;
+    };
+...
diff --git a/MAINTAINERS b/MAINTAINERS
index 0dc31ace8161..4d64149c9065 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -2530,6 +2530,7 @@ ARM/PHYTIUM SOC SUPPORT
 M:	Chen Baozi <chenbaozi@phytium.com.cn>
 S:	Maintained
 W:	https://www.phytium.com.cn
+F:	Documentation/devicetree/bindings/dma/phytium,ddma.yaml
 F:	Documentation/devicetree/bindings/gpio/phytium,pe220x-edac.yaml
 F:	Documentation/devicetree/bindings/gpio/phytium,gpio.yaml
 F:	Documentation/devicetree/bindings/gpio/phytium,sgpio.yaml
-- 
2.47.0

